js/block.js

```diff
@@ -3656,6 +3656,17 @@ class Block {
                     '" />';
                 labelElem.classList.add("hasKeyboard");
                 this.label = docById("numberLabel");
+
+                    // Add an input event listener to enforce the 4-digit limit
+        this.label.addEventListener("input", function () {
+        const inputValue = this.value;
+        if (inputValue.length > 6) {
```

Comment:
Do we want to let neg. numbers be up to 7 characters long, where one of the "digits" is the minus sign?

Reply:
Also, if it is a float, we really only care about the digits to the left of the ".".

Reply:
Sir, for negative numbers, users will be able to input a number to a minimum of -99999 as "-" would also be counted in length. For float, I will first convert it to an integer then take the input